## Coding our first neurons
#### A single neuron 

In [1]:
inputs = [1,2,3]
weights = [0.2,0.8,-0.5] 
bias = 2 

In [4]:
output = inputs[0]*weights[0] + inputs[1]*weights[1] + inputs[2]*weights[2] + bias

In [5]:
print(output)

2.3


### Visualization for the above code 
https://nnfs.io/bkr

In [6]:
## Let's change to 4 inputs 
inputs = [1.0,2.0,3.0,2.5] 
weights = [0.2,0.8,-0.5,1.0] 
bias = 2 

In [10]:
output = (inputs[0]*weights[0] + inputs[1]*weights[1] + inputs[2]*weights[2] + inputs[3]*weights[3] + bias) 

In [11]:
print(output)

4.8


![alt text](neurons.png "Title")

### Layers of Neurons 
![alt text](Neurons2.png "Title")

In [15]:
inputs = [1,2,3,2.5] 

weights1 = [0.2,0.8,-0.5,1]
weights2 = [0.5,-0.91,0.26,-0.5]
weights3 = [-0.26,-0.27,0.17,0.87] 
bias1 = 2 
bias2 = 3 
bias3 = 0.5 


In [16]:
output = [
    #Neuron 1: 
    inputs[0]*weights1[0] + 
    inputs[1]*weights1[1] + 
    inputs[2]*weights1[2] +
    inputs[3]*weights1[3]+ bias1,

    #Neuron 2 : 
    inputs[0]*weights2[0] + 
    inputs[1]*weights2[1] + 
    inputs[2]*weights2[2] +
    inputs[3]*weights2[3] + bias2,

    #Neuron 3 : 
    inputs[0]*weights3[0] +
    inputs[1]*weights3[1] + 
    inputs[2]*weights3[2] + 
    inputs[3]*weights3[3] + bias3 


]

In [17]:
print(output)

[4.8, 1.21, 2.385]


1. **Neuron 1 will provide output of 4.8**
2. **Neuron 2 will provide output of 1.21**
3. **Neuron 3 will provide output of 2.385** 
* This can serve as input for the neurons for the next layers 

Visuals for above code : https://nnfs.io/mxo

## This is the example of fully connected neural network 